### ISSUE TITLE
Lab Report 2 Feedback

### ISSUE THREAD

**Issue by eliasfang** [AngeliaZddl/CSE15L_Lab_Report#2]
Here is your feedback for Lab Report 2:

- [x] Correct screenshot of code diff page
- [x] Correct failure inducing test file - 1
- [x] Correct failure inducing test file - 2
- [x] Correct failure inducing test file - 3
- [x] Screenshot of symptom output that caused the failure for test file - 1
- [x] Screenshot of symptom output that caused the failure for test file - 2
- [x] Screenshot of symptom output that caused the failure for test file - 3
- [x] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 1
- [ ] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 2
- [ ] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 3
- [x] Correct markdown formatting
- [ ] Lab report is on its own page, not the index page
- [x] Text is formatted in a way that was easy for grader to read

Additional comments from your grader:
- Please include how the test files caused the symptom and helped you find the bug
- Please have the links to your test files go to the actual .md file instead of a .html page
- Your Lab Report 2 is in a completely separate repo - it should actually be in this repo!
- I like the theme!

--------

If you don't have all the checkmarks above, fix the issues described, and then submit a [regrade request on Gradescope](https://help.gradescope.com/article/8hchz9h8wh-student-regrade-request). Use this template for your regrade request:

- Link to this issue
- Link to your published lab report page

**Comment by AngeliaZddl** [comment]
Hi, I just want to check if this what you are asking for the second change: 
Before the change, the code cannot skip the empty links. It will add an null in the list, a comma before the actual link but we only want to the actual links. Then I add a if condition to check if the link is empty, which means nothing in parens. If not, nothing will be added to the list.

But I'm not sure what I missed for the third change. May I have more hits about it?

**Comment by eliasfang** [comment]
Yep, that description looks better! For a third change, I'd recommend looking at the test file that helped you see the bug in the first place and how it caused the symptoms to show up. If you still get stuck, please visit tutor hours but I hope that helps!

**Comment by AngeliaZddl** [comment]
Hi, may I ask if this will satisfy the the third change?
Before the change, the code will run an error if there is an open bracket following the link without a close one. To fix this error and return the links in the output, I add a if condition to check if there is a close bracket following the open bracket. If not, break the while loop and return the links before this open bracket.

**Comment by eliasfang** [comment]
Close. You should also explain how the symptoms you observed tells you what the error is. For future reference, please ask questions on either Piazza or through tutor hours on Autograder and not on Github Issues :)

**Comment by eliasfang** [comment]
Here is your feedback for Lab Report 2 Regrade Request:

- [x] Correct screenshot of code diff page
- [x] Correct failure inducing test file - 1
- [x] Correct failure inducing test file - 2
- [x] Correct failure inducing test file - 3
- [x] Screenshot of symptom output that caused the failure for test file - 1
- [x] Screenshot of symptom output that caused the failure for test file - 2
- [x] Screenshot of symptom output that caused the failure for test file - 3
- [x] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 1
- [x] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 2
- [x] Correct description for the relationship between the bug, the symptom, and the failure-inducing input for file - 3
- [x] Correct markdown formatting
- [x] Lab report is on its own page, not the index page
- [x] Text is formatted in a way that was easy for grader to read

Additional comments from your grader:
- Great job!
